crates/bevy_ecs/src/entity/map_entities.rs

```diff
@@ -23,36 +23,57 @@ pub trait MapEntities {
     fn map_entities(&mut self, entity_map: &EntityMap) -> Result<(), MapEntitiesError>;
 }
 
+/// A hash map adapter for mapping entities.
```

Comment:
I'd like a little more context on where this could be useful.

Reply:
It's okay, it comes with time! Here's my crack at it.

```suggestion
/// A mapping from one set of entities to another.
///
/// The API generally follows [`HashMap`], but each [`Entity`] is returned by value, as they are [`Copy`].
///
/// This is typically used to coordinate data transfer between sets of entities, such as between a scene and the world or over the network.
/// This is required as [`Entity`] identifiers are opaque; you cannot and do not want to reuse identifiers directly. 
```

Reply:
I like it! Thank you!